sunpy/instr/tests/test_aia.py

```diff
@@ -16,6 +16,8 @@
 def test_aiaprep():
     # Test that header info for the map has been correctly updated
     # Check all of these for Map attributes and .meta values?
+    # Check array shape
+    assert prep_map.data.shape == original.data.shape
```

Comment:
I would prefer an error than a warning, as warning could be disable... and this should be a "not to do". 
I'm not too familiar with AIA operations to know whether they will have images with different sizes in some situations (or not squared). I know sometimes they've [pointed out](http://sdoisgo.blogspot.co.uk/2011/07/sdo-weekly-report-for-july-22-2011.html) but I don't know how often that happens and how we should prep these images. In any case I would say that the size of both axis being the same and that the observing range is larger than the sun size could be a good start.